crates/sample-kmdf-driver/Makefile.toml

```diff
@@ -1,4 +1,5 @@
 extend = "../wdk-build/rust-driver-makefile.toml"
 
-[env]
-WDK_BUILD_ADDITIONAL_INFVERIF_FLAGS = "/msft"
+[tasks.infverif]
+dependencies = ["wdk-samples-setup"]
+condition = { env_not_set = ["WDK_INF_USING_NEW_VERSION"] }
```

Comment:
I guess it's a tradeoff of jank.  From my perspective having this script in each sample TOML feels less clean than just having an environment variable they can consume.  I like your more explicit messages here though and you're correct that the fewer environment variables we have floating around the better.

Maybe we can split this into a "samples.toml" that then gets extended/imported by samples instead, and also use that for any future sample-specific logic we wind up needing? Then we don't have a big script block in every sample TOML.

Reply:
I actually love the idea of of a `samples-makefile.toml` that all samples can extend. It also looks like cargo make supports extending multiple files and will honour the loading order you specify. We could put the whole infverif override block in 
the `samples-makefile.toml` and just extend `samples-makefile.toml` after `rust-driver-makefile.toml`, so that the samples `infverif` tasks are all overidden. Also love that we can put any further samples-specific extension points there too. 

Assuming we want to put `samples-makefile.toml` in this repo (which i think is reasonable since it requires explicit inclusion on the consumer side) and have it extended in both the samples in this repo and the samples in the dedicated samples repo, we would need to package and distribute it the same way we do for `rust-driver-makefile.toml`. 

This would mean creating a new function in `wdk_build::cargo_make`, called something like `load_samples_makefile`. This new function could reuse most of the logic in `load_rust_driver_makefile` (this could get refactored out into a common function). The sample repo's load_script block would now look like:
```toml
load_script = '''
#!@rust
//! ```cargo
//! [dependencies]
//! wdk-build = "0.2.0"
//! ```
#![allow(unused_doc_comments)]

wdk_build::cargo_make::load_rust_driver_makefile()?;
wdk_build::cargo_make::load_samples_makefile()? // This line is new!
'''
```

Reply:
Since this change in the samples repo depends on a new version of this crate, I haven't opened a PR there yet.  Is this OK?

Reply:
Created a PR in the samples repo for once this crate is updated and released: https://github.com/microsoft/Windows-rust-driver-samples/pull/18